core/src/cats/update_bareos_tables.in

```diff
@@ -22,175 +22,98 @@
 #
 # This script will update a BAREOS database to the latest version.
 #
-
+set +e
+set +u
 #
 # Source the Bareos config functions.
 #
-. @scriptdir@/bareos-config-lib.sh
+. "@scriptdir@"/bareos-config-lib.sh
 
-db_name="${db_name:-`get_database_name @db_name@`}"
-db_user="${db_user:-`get_database_user @db_user@`}"
-db_version=`get_database_version`
-bareos_sql_ddl=`get_database_ddl_dir`
+db_name="${db_name:-$(get_database_name @db_name@)}"
+db_user="${db_user:-$(get_database_user @db_user@)}"
+db_version=$(get_database_version)
+bareos_sql_ddl="$(get_database_ddl_dir)"
 temp_sql_schema="/tmp/tables.sql.$$"
-default_db_type=`get_database_driver_default`
-working_dir=`get_working_dir`
 
 SHOW_WARNING="yes"
 
-if [ "$1" = "-f" ]; then
-   FORCE="yes"
-   shift || true
-fi
-
-#
-# See if the first argument is a valid backend name.
-# If so the user overrides the default database backend.
-#
 if [ $# -gt 0 ]; then
-   case $1 in
-      sqlite3)
-         db_type=$1
-         shift
-         ;;
-      mysql)
-         db_type=$1
-         shift
-         ;;
-      postgresql)
-         db_type=$1
-         shift
-         ;;
-      *)
-         ;;
-   esac
-fi
-
-#
-# If no new db_type is gives use the default db_type.
-#
-if [ -z "${db_type}" ]; then
-   db_type="${default_db_type}"
-fi
-
-echo "Updating ${db_type} tables"
-
-bindir=`get_database_utility_path ${db_type}`
-if [ ! -z "${bindir}" ]; then
-   PATH="$bindir:$PATH"
+  handle_database_scripts_command_line_parameter "$*"
```

Comment:
This should work, howver, it ignores cases like `update_bareos_tables -f mysql`, as only the first parameter is checked.